packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNode.ts

```diff
@@ -455,36 +442,9 @@ export class SummarizerNode implements IRootSummarizerNode {
 
 				// Fetch the latest snapshot and refresh state from it. Note that we need to use the reference sequence number
 				// of the fetched snapshot and not the "summaryRefSeq" that was passed in.
-				const { snapshotTree, snapshotRefSeq: fetchedSnapshotRefSeq } =
-					await fetchLatestSnapshot();
-
-				// Possible re-entrancy. We may have updated latest summary state while fetching the snapshot. If the fetched
-				// snapshot is older than the latest tracked summary, ignore it.
-				if (this.referenceSequenceNumber >= fetchedSnapshotRefSeq) {
-					eventProps.latestSummaryUpdated = false;
-					event.end(eventProps);
-					return { latestSummaryUpdated: false };
-				}
+				await fetchLatestSnapshotAndClose();
```

Comment:
Instead of this code deciding to close the container, maybe it can just return that it did not find a pending summary and the caller can then do the fetch and close. That way, this function does not even need to be passed here and all the logic is in one place - container runtime.

Reply:
Sounds good, let's loop this change into the `next` branch when we do the whole removal of the refresh from server code flow.

Reply:
AB#5152